File: dart/test/ftn_test.dart
import 'package:idkollen_client/idkollen_client.dart';
import 'package:test/test.dart';

import 'mock_client.dart';

void main() {
  late MockClient mock;
  late IdkollenClient client;

  setUp(() {
    mock = MockClient();
    client = IdkollenClientBuilder('cid', 'sec').baseUrl('https://x.test').httpClient(mock).build();
  });

  test('auth pending', () async {
    mock.enqueue(200, {
      'status': 'PENDING',
      'id': 'f1',
      'url': 'https://login',
    });
    final result = await client.ftn.auth(const FtnAuthRequest());
    expect(result, isA<FtnPending>());
    expect((result as FtnPending).url, 'https://login');
  });

  test('age verification completed', () async {
    mock.enqueue(200, {
      'status': 'COMPLETED',
      'id': 'av1',
      'ageVerified': true,
    });
    final result = await client.ftn.ageVerificationStatus('av1');
    expect(result, isA<AgeVerificationCompleted>());
    expect((result as AgeVerificationCompleted).ageVerified, true);
  });
}

File: ruby/test/ftn_test.rb
# frozen_string_literal: true

require_relative 'test_helper'
require 'json'

class FtnTest < Minitest::Test
  def setup
    @mock = MockAdapter.new
    @client = Idkollen::ClientBuilder.new('cid', 'sec').http_client(@mock.connection).build
  end

  def test_auth_pending
    @mock.enqueue(200, { 'status' => 'PENDING', 'id' => 'f1', 'url' => 'https://login' })
    result = @client.ftn.auth(Idkollen::Ftn::AuthRequest.new)
    assert_kind_of Idkollen::Ftn::Pending, result
    assert_equal 'https://login', result.url
  end

  def test_age_verification_completed
    @mock.enqueue(200, { 'status' => 'COMPLETED', 'id' => 'av1', 'ageVerified' => true })
    result = @client.ftn.age_verification(Idkollen::AgeVerification::Request.new(min_age: 18))
    assert_kind_of Idkollen::AgeVerification::Completed, result
    assert_equal true, result.age_verified
  end
end
